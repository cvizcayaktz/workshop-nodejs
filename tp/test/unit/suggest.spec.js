const React = require('react');
const renderer = require('react-test-renderer');
const Suggest = require('../../views/components/Suggest');

describe('Suggest (snapshots)', () => {
  const component = renderer.create(<Suggest query="iph" />);

  it('Suggest renders without results', () => {
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })

  it('Suggest renders with results [change state]', () => {
    const instance = component.getInstance();
    instance.setState({
      results:['iphone, iphone2']
    })
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })

});
