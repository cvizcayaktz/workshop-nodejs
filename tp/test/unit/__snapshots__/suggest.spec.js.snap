// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggest (Snapshot) Suggest renders with results [Change State] 1`] = `
<div>
  <ul
    className="suggest"
  >
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone"
      >
        iphone
      </a>
    </li>
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone 2"
      >
        iphone 2
      </a>
    </li>
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone 3"
      >
        iphone 3
      </a>
    </li>
  </ul>
</div>
`;

exports[`Suggest (Snapshot) Suggest renders with results [componentWillReceiveProps] 1`] = `
<div>
  <ul
    className="suggest"
  >
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone"
      >
        iphone
      </a>
    </li>
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone 2"
      >
        iphone 2
      </a>
    </li>
    <li
      className="suggest__item"
    >
      <a
        href="/app/search?q=iphone 3"
      >
        iphone 3
      </a>
    </li>
  </ul>
</div>
`;

exports[`Suggest (Snapshot) Suggest renders without results 1`] = `<div />`;
